src/Evolution/Systems/NewtonianEuler/PrimitiveVariables.cpp

```diff
@@ -0,0 +1,65 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/NewtonianEuler/PrimitiveVariables.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace NewtonianEuler {
+template <size_t Dim, typename DataType>
+tnsr::I<DataType, Dim> compute_velocity_from_conserved_quantities(
+    const Scalar<DataType>& density, const tnsr::I<DataType, Dim>& momentum,
+    const Scalar<DataType>& /*energy*/) noexcept {
+  auto velocity = make_with_value<tnsr::I<DataType, Dim>>(density, 0.0);
+  auto one_over_density = make_with_value<Scalar<DataType>>(density, 0.0);
+
+  get(one_over_density) = 1.0 / get(density);
```

Comment:
This doesn't have to be a Scalar.  If you make it a plain `DataType` (don't use `auto`) then you don't need the `make_with_value` and it can be const.